### DISCUSSION TITLE
Proposal: Support multilingual transcripts

### DISCUSSION THREAD

**Discussion by ryan-lp** [Podcastindex-org/podcast-namespace#483] [Enhancement Proposal]
Some podcasts are multilingual, where each episode might use a different language, or even where a single episode may switch between multiple languages.

It is already possible to list multiple languages in the channel of the RSS feed (e.g. `<language>en,es</language>` on the channel), and perhaps there should also be a similar optional tag on each item that defaults to the channel language, because that may be helpful when each episode is in a different language.

But when a single episode contains multiple languages, we also need a way to tag which text belong to which language within the transcript.

I am not sure if there is an obvious way to do it in every format, but for JSON, we can add an optional `language` property to each segment which defaults to the item's language in the RSS feed, as follows:

```json
    {
      "speaker": "Darth Vader",
      "startTime": 0.5,
      "endTime": 0.75,
      "body": "I",
      "language": "en"
    }
```

For WebVTT, maybe this information could be placed in a [comment](https://www.w3.org/TR/webvtt1/#introduction-comments).
For SRT, maybe this information could be encoded in parentheses or some other type of brackets.
For HTML, maybe this can use the [lang](https://html.spec.whatwg.org/multipage/dom.html#the-lang-and-xml:lang-attributes) attribute.

**Comment by saerdnaer** [comment]
why not `lang` instead of `language`?

**Comment by ryan-lp** [comment]
The RSS spec defines a `language` tag, the HTML spec defines a `lang` attribute. I have followed the existing style in each format, where for JSON, it uses unabbreviated words (otherwise we could certainly make most of the other JSON names shorter, but there is no difference after compression.)

**Comment by ryan-lp** [comment]
> perhaps there should also be a similar optional tag on each item that defaults to the channel language, because that may be helpful when each episode is in a different language.

For this part of the problem, we can actually use the existing [xml:lang](https://www.w3.org/TR/REC-xml/#sec-lang-tag) attribute:

```xml
<item xml:lang="pt">
  ...
</item>
<item xml:lang="es">
  ...
</item>
```

Technically these existing `language`/`lang` tags/attributes are specified to hold only one language, although in practice creators of multilingual podcasts do use comma-delimited lists in the `language` tag to hold multiple languages
